src/Azos/Apps/Volatile/ObjectStoreDaemon.cs

```diff
@@ -547,149 +521,141 @@ protected override void DoWaitForCompleteStop()
       WriteLog(MessageType.Trace, nameof(DoWaitForCompleteStop), "Exiting");
     }
 
-
     #endregion
 
-
     #region .pvt
 
+    private Bucket getBucket(Guid key)
+    {
+      var idx = (key.GetHashCode() & CoreConsts.ABS_HASH_MASK) % m_BucketCount;
+      return m_Buckets[idx];
+    }
 
-        private Bucket getBucket(Guid key)
+    private void threadSpin()
+    {
+      try
+      {
+        while (Running)
         {
-          var idx = (key.GetHashCode() & CoreConsts.ABS_HASH_MASK) % m_BucketCount;
-          return m_Buckets[idx];
-        }
+          visit(false);   //todo need to make sure that this visit() never leaks anything otherwise thread may crash the whole service
+          m_Trigger.WaitOne(200);
+        }//while
+
+        visit(true);
+      }
+      catch (Exception e)
+      {
+        WriteLog(MessageType.Emergency, nameof(threadSpin), "Leaked exception: " + e.ToMessageWithType(), e);
+      }
+
+      WriteLog(MessageType.Trace, nameof(threadSpin), "Exiting");
+    }
 
-        private void threadSpin()
+    public void visit(bool stopping)
+    {
+      var now = App.LocalizedTime;
+      for (var i = 0; i < m_BucketCount; i++)
+      {
+        var bucket = m_Buckets[i];
+        if (stopping || bucket.LastAcquire.AddMilliseconds(MUST_ACQUIRE_INTERVAL_MS) < now)
+          lock (bucket) write(bucket);
+        else
+          if (Monitor.TryEnter(bucket))
         {
           try
           {
-              while (Running)
-              {
-                visit(false);   //todo need to make sure that this visit() never leaks anything otherwise thread may crash the whole service
-                m_Trigger.WaitOne(200);
-              }//while
-
-              visit(true);
+            write(bucket);
           }
-          catch(Exception e)
+          finally
           {
-              WriteLog(MessageType.Emergency, nameof(threadSpin),"Leaked exception: " + e.ToMessageWithType(), e);
+            Monitor.Exit(bucket);
```

Comment:
Where is WiteLog() and catch?  Is it a merge artifact or was that deleted from code?